src/dali/plugin/pair_converter/csv_reader/kraken_csv_pricing.py

```diff
@@ -0,0 +1,205 @@
+_QUERY: str = "q"
+
+_MS_IN_SECOND: int = 1000
+
+
+class KrakenCsvPricing:
+
+    __KRAKEN_OHLCVT: str = "Kraken.com_CSVOHLCVT"
+
+    __TIMEOUT: int = 30
+    __THREAD_COUNT: int = 6
+
+    __TIMESTAMP_INDEX: int = 0
+    __OPEN: int = 1
+    __HIGH: int = 2
+    __LOW: int = 3
+    __CLOSE: int = 4
+    __VOLUME: int = 5
+    __TRADES: int = 6
+
+    __DELIMITER: str = ","
+
+    def __init__(
+        self,
+        google_api_key: str,
+    ) -> None:
+
+        self.__google_api_key: str = google_api_key
+        self.__logger: logging.Logger = create_logger(self.__KRAKEN_OHLCVT)
+        self.__session: Session = requests.Session()
+
+    def get_historical_bars_for_pair(self, base_asset: str, quote_asset: str) -> Dict[datetime, HistoricalBar]:
+        bars: Dict[datetime, HistoricalBar] = {}
+        base_file: str = f"{base_asset}_OHLCVT.zip"
+
+        self.__logger.debug(base_file)
+
+        with ZipFile(BytesIO(self._google_file_to_bytes(base_file))) as zipped_ohlcvt:
+            self.__logger.debug(zipped_ohlcvt.namelist())
+            all_timespans_for_pair: List[str] = [x for x in zipped_ohlcvt.namelist() if x.startswith(f"{base_asset}{quote_asset}_")]
+
+            if len(all_timespans_for_pair) == 0:
+                self.__logger.debug("Market not found in Kraken files. Skipping file read.")
+                return bars
+
+            with ThreadPool(self.__THREAD_COUNT) as pool:
+                processing_result_list: List[Dict[int, List[HistoricalBar]]] = pool.starmap(
+                    self._process_file, zip(all_timespans_for_pair, repeat(zipped_ohlcvt))
+                )
+
+        # Sort the bars by duration, largest duration first
+        sorted_bars: Iterable[Dict[int, List[HistoricalBar]]] = reversed(sorted(processing_result_list, key=lambda x: list(x.keys())[0]))
+
+        timed_bars: Dict[datetime, HistoricalBar] = {}
+
+        # Start with longest duration and replace it with smaller durations
+        for duration_bars in sorted_bars:
+            bars_for_duration = list(duration_bars.values())[0]
+            for hbar in bars_for_duration:
+                # create keys for every minute starting with longest duration
+                start_epoch: int = int(hbar.timestamp.timestamp())
+                end_epoch: int = int((hbar.timestamp + timedelta(minutes=list(duration_bars.keys())[0])).timestamp())
+                while start_epoch < end_epoch:
+                    timed_bars[datetime.fromtimestamp(start_epoch, timezone.utc)] = hbar
+                    start_epoch += 60
+
+        return timed_bars
+
+    # isolated in order to be mocked
+    def _google_file_to_bytes(self, file_name: str) -> bytes:
+        params: Dict[str, Any] = {
+            _QUERY: f"'{_KRAKEN_FOLDER_ID}' in parents and name = '{file_name}'",
+            _API_KEY: self.__google_api_key,
+        }
+        try:
+            response: Response = self.__session.get(_GOOGLE_APIS_URL, params=params, timeout=self.__TIMEOUT)
```

Comment:
Add a comment pointing to the comments you added above. We need to document any networking code that is not immediately recognizable as legit (e.g. any non-ccxt-based networking code), to ensure people who read the code understand nothing illegitimate is happening.